### ISSUE TITLE
Is it possible to take control of status bar taps on iOS?

### ISSUE THREAD

**Issue by mark8044** [NamanShergill/flutter_scroll_to_top#13]
Right now there are problems with nested Scaffolds and detecting status bar taps for scroll to top.  This would solve that, but the button on the top right is not as fluid appearing on iOS.  Is it possible to modify this project to simply take control of the status bar (a transparent space over the status bar?) to mimic native iOS function?

**Comment by NamanShergill** [comment]
That is a bit out of the scope of the library, have you had a look at this?

https://pub.dev/packages/scrolls_to_top

Also, I’m not sure if you were referring to issue #11 here, but I have published a pre release to fix that. 

https://github.com/NamanShergill/flutter_scroll_to_top/issues/11#issuecomment-1127318648

**Comment by mark8044** [comment]
> That is a bit out of the scope of the library, have you had a look at this?
> 
> https://pub.dev/packages/scrolls_to_top
> 
> Also, I’m not sure if you were referring to issue #11 here, but I have published a pre release to fix that.
> 
> [#11 (comment)](https://github.com/NamanShergill/flutter_scroll_to_top/issues/11#issuecomment-1127318648)

Yes, Ive tried that package but it doesn't work very well unfortunately.  

And yes x2, your #11 fix seems to be exactly ideal/perfect for dealing with nested Scaffolds.  

That is why I was hoping this package could be adapted so that the button could be alternatively transparent and wider to cover the very top status bar area, thus mimicking native iOS behavior.  Would be amazing :)

Thanks

**Comment by NamanShergill** [comment]
Hmm, I’ll have to research a bit on how flutter handles status bar taps, there’s some relevant discussion here: https://github.com/flutter/flutter/issues/85603#issuecomment-907757258

I would’ve suggested using the `promptReplacementBuilder`, but I don’t think there’s any way to overlay a widget on the status bar.

If I’m able to think of a way to incorporate it, will work on doing so.

**Comment by NamanShergill** [comment]
Just curious, now that the issue with primaryscrollcontrollers is fixed, why don’t you just use that? PrimaryScrollController has this built in with the status bar taps scrolling it to top, so not passing a scrollcontroller to a scrollview will automatically give you this behaviour.

**Comment by mark8044** [comment]
> Just curious, now that the issue with primaryscrollcontrollers is fixed, why don’t you just use that? PrimaryScrollController has this built in with the status bar taps scrolling it to top, so not passing a scrollcontroller to a scrollview will automatically give you this behaviour.

PSC only partially fixes the problem.  An issue comes up with nested Scaffolds, as people use with Persistent Bottom Navigation, in which case if you have a nested scaffold, the outer container scaffold takes the tap and nothing happens.

Its a much bigger problem that needs to be addressed, but certainly outside the scope of your project :)

**Comment by NamanShergill** [comment]
I will research on this, if it’s something I could figure out on how to incorporate organically, will do so.

**Comment by mark8044** [comment]
> I will research on this, if it’s something I could figure out on how to incorporate organically, will do so.

No worries, I appreciate the attempt.  I was hoping it was something simple like adapting your work to put a transparent container over the top of the screen, but it sounds like its more involved, so no big deal.  👍
